projects/rero/ng-core/src/lib/record/editor/array-type/array-type.component.ts

```diff
@@ -0,0 +1,130 @@
+/*
+ * Invenio angular core
+ * Copyright (C) 2019 RERO
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, version 3 of the License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { Component, OnInit } from '@angular/core';
+import { FieldArrayType, FormlyFieldConfig } from '@ngx-formly/core';
+
+@Component({
+  selector: 'ng-core-editor-formly-array-type',
+  templateUrl: 'array-type.component.html'
+})
+export class ArrayTypeComponent extends FieldArrayType implements OnInit {
+
+  /**
+   * Are the children of type object?
+   */
+  isChildrenObject = false;
+
+  /**
+   * Component initialisation
+   */
+  ngOnInit() {
+    this.isChildrenObject = this.field.fieldArray.type === 'object';
+  }
+
+  /**
+   * Is a new element can be added?
+   * @returns boolean, true if a new element can be inserted in the array
```

Comment:
Usually, it's better to type the method return than specify @returns in the doc block.